### PR TITLE
modify model, query and payloads

### PR DESCRIPTION
**Pull request by ramkrishs**

### CODE REVIEW

File: src/main/java/com/chalktool/api/AttendanceApi.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 public interface AttendanceApi {
 
-    String postAttendance(String courseId, String date, List<String> absenteeList);
+    String postAttendance(String courseId, String date, List<String> absenteeList, int presentCount, int totalCount);
```
Comment (muneerulhudha) [inline src/main/java/com/chalktool/api/AttendanceApi.java]: do we really need to persist this in the DB as well? Why can't we just do the count ourselves.

File: src/main/java/com/chalktool/dao/ClassDao.java [hunk 2]
```diff
@@ -16,10 +16,13 @@
...
 
-    @SqlUpdate("insert into class(class_id ,course_id, date) " +
-            "values(:class_id,:course_id, :date) ")
+    @SqlUpdate("insert into class(class_id ,course_id, date, date_in_words, present_count, total_count) " +
+            "values(:class_id, :course_id, :date, :date_in_words, :present_count, :total_count) ")
     void addClass(@Bind("class_id") String classId,
                    @Bind("course_id") String courseId,
-                   @Bind("date")Date date);
+                   @Bind("date") Date date,
+                  @Bind("date_in_words") String dateInWords,
```
Comment (muneerulhudha) [inline src/main/java/com/chalktool/dao/ClassDao.java]: Why is date in words necessary? Can't we just use getDayOfWeek() method from LocalDate java util?

File: src/main/java/com/chalktool/util/Iso8601Utils.java [hunk 3]
```diff
@@ -0,0 +1,342 @@
+package com.chalktool.util;
```
Comment (muneerulhudha) [inline src/main/java/com/chalktool/util/Iso8601Utils.java]: What is this util class? Why do we need it? Can't we just use a class thats part of a library? Or add this as a dependency?

Comment (muneerulhudha) [reply src/main/java/com/chalktool/util/Iso8601Utils.java]: I think its part of this dependency:

<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-databind</artifactId>
    <version>2.10.2</version>
</dependency>

File: src/main/java/com/chalktool/api/AttendanceApi.java (hunk 1 above)
Comment (ramkrishs) [reply src/main/java/com/chalktool/api/AttendanceApi.java]: finding total students and getting absentee count will be two different calls instead Now when we fetch pastattendance list for a course we can store these values in the recycle viewer and pass it to the next screen directly.

File: src/main/java/com/chalktool/dao/ClassDao.java (hunk 2 above)
Comment (ramkrishs) [reply src/main/java/com/chalktool/dao/ClassDao.java]: In Utils we extract "Tuesday, 13 March" from the date once and saving it in two formats if we just passing the date and computing this for each item in the list it is time-consuming in the app side so whenever we insert an attendance we compute once.

File: src/main/java/com/chalktool/util/Iso8601Utils.java (hunk 3 above)
Comment (ramkrishs) [reply src/main/java/com/chalktool/util/Iso8601Utils.java]: We have similar one in moshi  dependencies , what i did is copied from it and added extra time-based util functions. we want same util consistent in both app and server side that's why added this util da.

**Pull request merged**